### PR TITLE
fix(matchers): Don't sort failed matches when printing error message

### PR DESCRIPTION
**Pull request by mgaligniana**
Hi!

This PR removes the `_create_key_val_str` and with it the sorting when displaying matchers errors

Let me know any change required!

Thank you 

Fixes GH-704

### CODE REVIEW

**Comment by beliaev-maksim** [comment]
@markstory I think we introduced `_create_key_val_str` for the compatibility between python2 and python3

should we just drop it completely and just present JSON?

**Comment by beliaev-maksim** [comment]
@mgaligniana please add changelog file entry

**Comment by mgaligniana** [comment]
@beliaev-maksim thank you for your fast response! On it! :factory_worker:

**Comment by mgaligniana** [comment]
@beliaev-maksim Done! Let me know if message is OK (I'm not English native speaker). Feel free to suggest changes! Thank you!

**Comment by markstory** [comment]
> I think we introduced _create_key_val_str for the compatibility between python2 and python3 should we just drop it completely and just present JSON?

I think you're right on the origin of that method. We could likely drop it now that we're not supporting python 2 anymore.

**Comment by mgaligniana** [comment]
@markstory Do you want me to tackle in this pull request? Or in another ticket?

**Comment by beliaev-maksim** [comment]
@mgaligniana yes please

**Comment by markstory** [comment]
> Do you want me to tackle in this pull request? Or in another ticket?

Either works for me. Using this pull request would be simpler for us though.

**Comment by mgaligniana** [comment]
Done!

I had to update all test cases because now it shows inputs are strings in error messages.

For example in `query_param_matcher`
Before: `...doesn't match {array: [B, A, C]}` After: `...doesn't match: {'array': ['B', 'A', 'C']}`

File: CHANGES [hunk 1]
```diff
@@ -1,3 +1,9 @@
+0.25.2
+------
+
+* Fixed error messages when matches fail. See #704
+* Delete `_create_key_val_str` which was introduced for the compatibility between python2 (not supported anymore) and python3.
```
Comment (beliaev-maksim) [inline CHANGES:5]: I would not talk about internal method. Instead, please briefly describe what changed for the end user

**Review by beliaev-maksim — CHANGES_REQUESTED** [review]

**Comment by mgaligniana** [comment]
> I would not talk about internal method. Instead, please briefly describe what changed for the end user

Makes sense!

Ready, thanks for your help and patience!

**Review by beliaev-maksim — APPROVED** [review]

**Comment by beliaev-maksim** [comment]
looks good, thanks

**Comment by markstory** [comment]
There are a few failing tests that will need to be addressed before this could be merged.

**Comment by mgaligniana** [comment]
> There are a few failing tests that will need to be addressed before this could be merged.

Will star working on that! I'm going to use `tox` to can reproduce them! 👷 🚧

**Comment by mgaligniana** [comment]
Now it is! I missed a f-string and probably forgot run the tests before push 🤷, sorry!

**Comment by mgaligniana** [comment]
@markstory sorry for the direct ping! But could you re-run the tests? Thank you!

**Comment by markstory** [comment]
Thank you 🎉

**Pull request merged**